extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -764,36 +766,63 @@ function getPageLevelParameters_(win, doc, startTime, isSra) {
 }
 
 /**
- * @param {?Object<string, (!Array<string>|string)>} targeting
- * @param {?(!Array<string>|string)} categoryExclusions
+ * @param {?JsonObject} targeting
+ * @param {(?JsonValue|undefined)} categoryExclusions
  * @return {?string}
  * @private
  */
 function serializeTargeting_(targeting, categoryExclusions) {
   const serialized = targeting ?
       Object.keys(targeting).map(key => serializeItem_(key, targeting[key])) :
       [];
-  if (categoryExclusions) {
+  if (typeof categoryExclusions != 'undefined') {
     serialized.push(serializeItem_('excl_cat', categoryExclusions));
   }
   return serialized.length ? serialized.join('&') : null;
 }
 
 /**
  * @param {string} key
- * @param {(!Array<string>|string)} value
+ * @param {?JsonValue} value
  * @return {string}
  * @private
  */
 function serializeItem_(key, value) {
-  const serializedValue =
-    (Array.isArray(value) ? value : [value]).map(encodeURIComponent).join();
+  let serializedValue;
+  if (typeof value == 'string') {
+    serializedValue = encodeURIComponent(/** @type {string} */ (value));
+  } else if (
+      Array.isArray(value) && /** @type {!Array<?JsonValue>} */
+      (value).every(element => typeof element == 'string')) {
```

Comment:
What happens if one is not a string?  We drop all of them?  Is that the correct behavior?  Why not allow number for instance?